### PR TITLE
Add "No Scope" mod implementation for Catch

### PR DESCRIPTION
**Pull request by Semro**
Resolves https://github.com/ppy/osu/discussions/15128

Didn't come up with good mod namings, open for suggestions.

### CODE REVIEW

File: osu.Game/Rulesets/Mods/ModNoScope.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        protected BindableNumber<int> CurrentCombo;
+
+        protected IBindable<bool> IsBreakTime;
+
+        protected float ComboBasedAlpha;
+
+        [SettingSource(
+            "Hidden at combo",
+            "The combo count at which the cursor becomes completely hidden",
```
Comment (bdach) [inline osu.Game/Rulesets/Mods/ModNoScope.cs:41]: Weird for this to mention the "cursor" in the base implementation, while it's actually going to affect the catcher in catch.

File: osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Utils;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Rulesets.UI;
+using osu.Game.Utils;
+
+namespace osu.Game.Rulesets.Catch.Mods
+{
+    public class CatchModNoScope : ModNoScope, IApplicableToBeatmap, IApplicableToDrawableRuleset<CatchHitObject>
```
Comment (bdach) [inline osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs:16]: Why `IApplicableToDrawableRuleset<CatchHitObject>` and not `IUpdatableByPlayfield`? Would remove the `playfield.OnUpdate` subscription entirely.

File: osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        public override string Description => "Where's the catcher?";
+
+        public PeriodTracker BananaShowerPeriods;
```
Comment (bdach) [inline osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs:20]: Doesn't  need to be public.

File: osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            BananaShowerPeriods = new PeriodTracker(beatmap.HitObjects.OfType<BananaShower>().Select(b => new Period(b.StartTime - TRANSITION_DURATION, b.EndTime)));
```
Comment (bdach) [inline osu.Game.Rulesets.Catch/Mods/CatchModNoScope.cs:24]: I'm actually not sure whether I'd hide the catcher during showers. They're kind of like normal fruit and also will provide *some* visual feedback, won't they? Also controlling the banana capture path to maximise caught fruit is a thing in catch I think, so it's not like a spinner where you go as fast as possible.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/Mods/ModNoScope.cs (hunk 1 above)
Comment (Semro) [reply osu.Game/Rulesets/Mods/ModNoScope.cs]: Made separated descriptions for each mod, but this leads to duplication. Maybe it's better to change "cursor" -> "player" in the base implementation?

Comment (bdach) [reply osu.Game/Rulesets/Mods/ModNoScope.cs]: I don't mind duplications. Chances are the ruleset implementation may diverge in more ways than that anyhow.

**Comment by bdach** [comment]
Made small changes here. Looks good otherwise.

Not a catch player so probably going to tag in @ekrctb for a second review playability-wise. Unsure if https://github.com/ppy/osu-framework/pull/4880 was intended for use here? Either way I think updating to that can happen later because the pattern that that pull is looking to replace (and which is used here) is also used in many more places.

**Review by bdach — APPROVED** [review]

**Review by ekrctb — APPROVED** [review]
This is actually fun to play.

**Comment by bdach** [comment]
For clarity of information, I have the appropriate change for web (to support submission with this mod) prepped. Will test and PR it later today.

**Pull request merged**